plugin/storage/es/mappings/mapping_test.go

```diff
@@ -0,0 +1,316 @@
+			want := ""
+			if tt.esVersion == 7 {
+				want, err = mb.fixMapping("/" + tt.mapping + "-7.json")
+				require.NoError(t, err)
+			} else {
+				want, err = mb.fixMapping("/" + tt.mapping + ".json")
+				require.NoError(t, err)
+			}
+			assert.Equal(t, got, want)
+		})
+	}
+}
+
+func TestMappingBuilder_loadMapping(t *testing.T) {
+	tests := []struct {
+		name        string
+		indexPrefix string
+		useILM      bool
+	}{
+		{name: "/jaeger-span.json"},
+		{name: "/jaeger-service.json"},
+		{name: "/jaeger-span-7.json"},
+		{name: "/jaeger-service-7.json"},
+		{name: "/jaeger-dependencies.json"},
+		{name: "/jaeger-dependencies-7.json"},
+	}
+	for _, test := range tests {
+		mapping := loadMapping(test.name)
+		f, err := os.Open("./" + test.name)
+		require.NoError(t, err)
+		b, err := ioutil.ReadAll(f)
+		require.NoError(t, err)
+		assert.Equal(t, string(b), mapping)
+		_, err = template.New("mapping").Parse(mapping)
+		require.NoError(t, err)
+	}
+}
+
+func TestMappingBuilder_fixMapping(t *testing.T) {
+	tests := []struct {
+		name                    string
+		templateBuilderMockFunc func() *mocks.TemplateBuilder
+		err                     string
+	}{
+		{
+			name: "templateRenderSuccess",
+			templateBuilderMockFunc: func() *mocks.TemplateBuilder {
+				tb := mocks.TemplateBuilder{}
+				ta := mocks.TemplateApplier{}
+				ta.On("Execute", mock.Anything, mock.Anything).Return(nil)
+				tb.On("Parse", mock.Anything).Return(&ta, nil)
+				return &tb
+			},
+			err: "",
+		},
+		{
+			name: "templateRenderFailure",
+			templateBuilderMockFunc: func() *mocks.TemplateBuilder {
+				tb := mocks.TemplateBuilder{}
+				ta := mocks.TemplateApplier{}
+				ta.On("Execute", mock.Anything, mock.Anything).Return(errors.New("template exec error"))
+				tb.On("Parse", mock.Anything).Return(&ta, nil)
+				return &tb
+			},
+			err: "template exec error",
+		},
+		{
+			name: "templateLoadError",
+			templateBuilderMockFunc: func() *mocks.TemplateBuilder {
+				tb := mocks.TemplateBuilder{}
+				tb.On("Parse", mock.Anything).Return(nil, errors.New("template load error"))
+				return &tb
+			},
+			err: "template load error",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			mappingBuilder := MappingBuilder{
+				TemplateBuilder: test.templateBuilderMockFunc(),
+				Shards:          3,
+				Replicas:        5,
+				EsVersion:       7,
+				IndexPrefix:     "test",
+				UseILM:          true,
+			}
+			_, err := mappingBuilder.fixMapping("test")
+			if test.err != "" {
+				assert.EqualError(t, err, test.err)
+			} else {
+				assert.NoError(t, err)
+			}
+
+		})
+	}
+}
+
+func TestMappingBuilder_GetSpanServiceMappings(t *testing.T) {
+	type args struct {
+		shards      int64
+		replicas    int64
+		esVersion   uint
+		indexPrefix string
+		useILM      bool
+	}
+	tests := []struct {
+		name                       string
+		args                       args
+		mockNewTextTemplateBuilder func() es.TemplateBuilder
```

Comment:
I would just use `MappingBuilder` struct directly, there's no benefit in the indirections

Reply:
Dint get this.

Reply:
rather than defining a struct for tests and copying values into it, just use MappingBuilder struct directly when defining test cases.

Reply:
Not part of this commit. Will work on it now.